### PR TITLE
Fix the logic issue of handling the ov::device::properties in the ie core.

### PR DESCRIPTION
**Pull request by yangwang201911**
Update the logic of handling the ov:device::properties in the ie core.

Signed-off-by: Wang, Yang <yang4.wang@intel.com>

### Details:
 - fix the logic issue of handling ov::device::properties in the function _flatten_sub_properties()_.
 - Add corresponding test case.

### Tickets:
 - 94258

### CODE REVIEW

File: src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp [hunk 1]
```diff
@@ -1096,6 +1097,18 @@ TEST_P(OVClassLoadNetworkWithCorrectPropertiesTest, LoadNetworkWithCorrectProper
...
 }
 
+TEST_P(OVClassLoadNetworkAndCheckSecondaryPropertiesTest, LoadNetworkAndCheckSecondaryPropertiesTest) {
+    ov::Core ie = createCoreWithTemplate();
+    ov::CompiledModel model;
+    OV_ASSERT_NO_THROW(model = ie.compile_model(actualNetwork, target_device, configuration));
+    auto property = configuration.begin()->second.as<ov::AnyMap>();
+    auto actual = property.begin()->second.as<int32_t>();
+    ov::Any value;
+    OV_ASSERT_NO_THROW(value = model.get_property(ov::num_streams.name()));
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp:1107]: what if we set different number of streams for each device? should we add such tests? what AUTO will reports in this case?

Comment (yangwang201911) [reply src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp:1107]: More test cases implemented in the auto mock test.

File: src/inference/src/ie_core.cpp [hunk 2]
```diff
@@ -182,13 +182,15 @@ void allowNotImplemented(F&& f) {
...
+    bool isVirtualDev = device.find("AUTO") != std::string::npos || device.find("MULTI") != std::string::npos ||
+                        device.find("HETERO") != std::string::npos;
     for (auto item = result.begin(); item != result.end();) {
         auto parsed = parseDeviceNameIntoConfig(item->first);
         if (!item->second.is<ov::AnyMap>()) {
             item++;
             continue;
         }
-        if (device.find(parsed._deviceName) != std::string::npos) {
+        if (device == parsed._deviceName) {
```
Comment (songbell) [inline src/inference/src/ie_core.cpp:193]: does this cover GPU.* case?

Comment (yangwang201911) [reply src/inference/src/ie_core.cpp:193]: Do you mean the device properties in the config like {"GPU.1", "NUM_STREAMS 4"} could be passed to GPU plugin correctly?

**Review by ilya-lavrenov — APPROVED** [review]

File: src/inference/src/ie_core.cpp (hunk 2 above)
Comment (songbell) [reply src/inference/src/ie_core.cpp:193]: yes, I mean be passed as the secondary property to AUTO/MULTI, maybe we can update relevant document to list all the situations and expected results, like for me with 2 GPUs,  I want to know if -d MULTI:CPU, GPU -nstreams CPU:3,GPU.0:3,GPU.1:4  is a valid setting from benchmark, and if all the settings can take effect in each hw plugins

**Comment by yangwang201911** [comment]
hi @ilya-lavrenov , I got a GTEST Mock build error similar as this  [link](https://github.com/google/googletest/issues/2897) when using a defined MATHCER_P and fixed it by adding override in  MATHCER_P Macro referred this [commit](https://github.com/google/googletest/commit/3cddd56e195b516f449bea6dcd3edd4494195631#diff-7a61265c545849e22230ee77d23fb58a0d079f6455d1a7e41a17d6d63fe32b64).
Should I just create a PR to openvinotoolkit/googletest with this workaround?

**Comment by ilya-lavrenov** [comment]
> hi @ilya-lavrenov , I got a GTEST Mock build error similar as this [link](https://github.com/google/googletest/issues/2897) when using a defined MATHCER_P and fixed it by adding override in MATHCER_P Macro referred this [commit](https://github.com/google/googletest/commit/3cddd56e195b516f449bea6dcd3edd4494195631#diff-7a61265c545849e22230ee77d23fb58a0d079f6455d1a7e41a17d6d63fe32b64). Should I just create a PR to openvinotoolkit/googletest with this workaround?

Yes, let's go this way

**Comment by yangwang201911** [comment]
> hi @ilya-lavrenov , I got a GTEST Mock build error similar as this https://github.com/google/googletest/issues/2897 when using a defined MATHCER_P and fixed it by adding override in MATHCER_P Macro referred this [commit](https://github.com/google/googletest/commit/3cddd56e195b516f449bea6dcd3edd4494195631#diff-7a61265c545849e22230ee77d23fb58a0d079f6455d1a7e41a17d6d63fe32b64). Should I just create a PR to openvinotoolkit/googletest with this workaround?

> Yes, let's go this way

PR is ready to review. https://github.com/openvinotoolkit/googletest/pull/1

**Comment by ilya-lavrenov** [comment]
```
2022-10-28T06:09:17.9628076Z In file included from ../../openvino/thirdparty/gtest/gtest/googlemock/include/gmock/gmock.h:64,
2022-10-28T06:09:17.9629341Z                  from ../../openvino/src/tests/unit/auto/auto_load_network_properties_test.cpp:5:
2022-10-28T06:09:17.9632062Z ../../openvino/src/tests/unit/auto/auto_load_network_properties_test.cpp: In instantiation of ‘class MapContainsMatcherP<std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> > >::gmock_Impl<const std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >&>’:
2022-10-28T06:09:17.9636052Z ../../openvino/src/tests/unit/auto/auto_load_network_properties_test.cpp:36:1:   required from ‘MapContainsMatcherP<subMap_type>::operator testing::Matcher<T>() const [with arg_type = const std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >&; subMap_type = std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >]’
2022-10-28T06:09:17.9638388Z ../../openvino/src/tests/unit/auto/auto_load_network_properties_test.cpp:259:9:   required from here
2022-10-28T06:09:17.9642768Z ../../openvino/src/tests/unit/auto/auto_load_network_properties_test.cpp:36:11: error: ‘bool MapContainsMatcherP<subMap_type>::gmock_Impl<arg_type>::MatchAndExplain(typename testing::internal::ConstRef<arg_type>::type, testing::MatchResultListener*) const [with arg_type = const std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >&; subMap_type = std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >; typename testing::internal::ConstRef<arg_type>::type = const std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >&]’ can be marked override [-Werror=suggest-override]
2022-10-28T06:09:17.9646069Z    36 | MATCHER_P(MapContains, subMap, "Check if all the elements of the subMap are contained in the map.") {
2022-10-28T06:09:17.9648396Z       |           ^
2022-10-28T06:09:17.9649163Z cc1plus: all warnings being treated as errors
```
on azure precommit

**Comment by yangwang201911** [comment]
> ```
> 2022-10-28T06:09:17.9628076Z In file included from ../../openvino/thirdparty/gtest/gtest/googlemock/include/gmock/gmock.h:64,
> 2022-10-28T06:09:17.9629341Z                  from ../../openvino/src/tests/unit/auto/auto_load_network_properties_test.cpp:5:
> 2022-10-28T06:09:17.9632062Z ../../openvino/src/tests/unit/auto/auto_load_network_properties_test.cpp: In instantiation of ‘class MapContainsMatcherP<std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> > >::gmock_Impl<const std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >&>’:
> 2022-10-28T06:09:17.9636052Z ../../openvino/src/tests/unit/auto/auto_load_network_properties_test.cpp:36:1:   required from ‘MapContainsMatcherP<subMap_type>::operator testing::Matcher<T>() const [with arg_type = const std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >&; subMap_type = std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >]’
> 2022-10-28T06:09:17.9638388Z ../../openvino/src/tests/unit/auto/auto_load_network_properties_test.cpp:259:9:   required from here
> 2022-10-28T06:09:17.9642768Z ../../openvino/src/tests/unit/auto/auto_load_network_properties_test.cpp:36:11: error: ‘bool MapContainsMatcherP<subMap_type>::gmock_Impl<arg_type>::MatchAndExplain(typename testing::internal::ConstRef<arg_type>::type, testing::MatchResultListener*) const [with arg_type = const std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >&; subMap_type = std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >; typename testing::internal::ConstRef<arg_type>::type = const std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >&]’ can be marked override [-Werror=suggest-override]
> 2022-10-28T06:09:17.9646069Z    36 | MATCHER_P(MapContains, subMap, "Check if all the elements of the subMap are contained in the map.") {
> 2022-10-28T06:09:17.9648396Z       |           ^
> 2022-10-28T06:09:17.9649163Z cc1plus: all warnings being treated as errors
> ```
> 
> on azure precommit

This warning msg disappeared on my host when updated GTEST following this [PR](https://github.com/openvinotoolkit/googletest/pull/1). Could you help to review this PR?

**Review by wangleis — APPROVED** [review]

**Comment by wangleis** [comment]
Reviewed in weekly sync up meeting

**Pull request merged**